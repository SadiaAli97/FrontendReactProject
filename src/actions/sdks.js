import * as types from '../constants/actions';

export const setSelectedToggleOption = (option) => dispatch => {
 dispatch({
  type: types.SET_TOGGLE,
  payload: option,
 })
}

export const loadSdkData = (installedSdks, uninstalledSdks) => dispatch => {
    dispatch({
        type: types.LOAD_SDK_DATA,
        payload: {
            installedSdks,
            uninstalledSdks
        }
    })
}