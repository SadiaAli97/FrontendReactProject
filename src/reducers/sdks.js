import * as types from '../constants/actions';

export default (state = {}, action) => {
 switch (action.type) {
  case types.SET_TOGGLE:
      return {
   	    ...state,
   	   selectedToggleOption: action.payload,
      }
   case types.LOAD_SDK_DATA:
     console.log('action payload ', action.payload);
     const { installedSdks, uninstalledSdks } = action.payload;
     return {
        ...state,
        installedSdks,
        uninstalledSdks
     }
  default:
   return state
 }
}