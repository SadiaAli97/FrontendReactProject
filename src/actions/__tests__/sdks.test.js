import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { setSelectedToggleOption, loadSdkData } from '../sdks'; 

const middlewares = []
const mockStore = configureStore(middlewares)

const props = {
  sdks: {
    selectedToggleOption: 'isdks',
    installedSdks: [],
    uninstalledSdks: [],
  }
}

const store = mockStore({})

it('sdks action setSelectedToggleOption ', () => {
  expect(setSelectedToggleOption('uisdks')).toMatchSnapshot()

});

it('sdks action loadSdkData ', () => {
  const installedSdks = [
  {
        "id": "1",
        "name": "SDK 1",
        "categories": ["Backend"],
        "firstSeenDate": "2017-11-16T10:48:53.000-08:00",
        "lastSeenDate": "2019-07-16T13:24:39.000-07:00",
        "__typename": "Sdk"
   },
  ];
  const uninstalledSdks = [
    {
        "id": "1",
        "name": "Sdk 1",
        "categories": ["App Performance Management"],
        "firstSeenDate": "2017-11-16T10:48:53.000-08:00",
        "lastSeenDate": "2019-03-15T06:12:41.000-07:00",
        "__typename": "Sdk"
      },
  ];	
  expect(loadSdkData()).toMatchSnapshot()

});