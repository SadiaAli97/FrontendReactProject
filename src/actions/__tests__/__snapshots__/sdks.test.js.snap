// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sdks action loadSdkData  1`] = `[Function]`;

exports[`sdks action setSelectedToggleOption  1`] = `[Function]`;
